pkg/handlers/v1/scan.go

```diff
@@ -144,24 +146,37 @@ func fromDomain(source []domain.Finding) []ScanFinding {
 	return r
 }
 
-// Scan is a handler that manages scanning a host on-deman.
+// Scan is a handler that manages scanning a host on-demand.
 type Scan struct {
-	LogFn    domain.LogFn
-	Scanner  domain.Scanner
-	Producer domain.Producer
+	LogFn           domain.LogFn
+	Scanner         domain.Scanner
+	ScriptedScanner domain.ScriptedScanner
+	Producer        domain.Producer
 }
 
-func (h *Scan) handle(ctx context.Context, in ScanInput) (ScanOutput, error) {
+func (h *Scan) handleScan(ctx context.Context, in ScanInput) (ScanOutput, error) {
 	findings, err := h.Scanner.Scan(ctx, in.Host)
 	if err != nil {
 		return ScanOutput{}, err
 	}
 	return ScanOutput{Findings: fromDomain(findings)}, nil
 }
 
+func (h *Scan) handleScriptScan(ctx context.Context, in ScanInput) (ScanOutput, error) {
+	findings, err := h.ScriptedScanner.ScanWithScripts(ctx, in.Scripts, in.ScriptArgs, in.Host)
+	if err != nil {
+		return ScanOutput{}, err
+	}
+	return ScanOutput{Findings: fromDomain(findings)}, nil
+}
+
 // Handle is invoked on each request.
 func (h *Scan) Handle(ctx context.Context, in ScanInput) (interface{}, error) {
-	v, err := h.handle(ctx, in)
+	handler := h.handleScan
+	if len(in.Scripts) > 0 || len(in.ScriptArgs) > 0 {
```

Comment:
Question: is it valid to call with script args, but no scripts?

Reply:
Yes. There are several flags present in "stdlib" modules that might be passed without additional scripts. If no scripts are given then nmap runs a default set of scripts anyway.